src/Sylius/Bundle/ApiBundle/ApiPlatform/Factory/MergingExtractorResourceMetadataFactory.php

```diff
@@ -0,0 +1,168 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\ApiPlatform\Factory;
+
+use ApiPlatform\Core\Exception\ResourceClassNotFoundException;
+use ApiPlatform\Core\Metadata\Extractor\ExtractorInterface;
+use ApiPlatform\Core\Metadata\Resource\Factory\ResourceMetadataFactoryInterface;
+use ApiPlatform\Core\Metadata\Resource\ResourceMetadata;
+
+
+final class MergingExtractorResourceMetadataFactory implements ResourceMetadataFactoryInterface
+{
+    /** @var ExtractorInterface */
+    private $extractor;
+
+    /** @var ?ResourceMetadataFactoryInterface  */
+    private $decorated;
+
+    /** @var array */
+    private $defaults;
+
+    public function __construct(ExtractorInterface $extractor, ResourceMetadataFactoryInterface $decorated = null, array $defaults = [])
+    {
+        $this->extractor = $extractor;
+        $this->decorated = $decorated;
+        $this->defaults = $defaults + ['attributes' => []];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function create(string $resourceClass): ResourceMetadata
+    {
+        $parentResourceMetadata = null;
+        if ($this->decorated) {
+            try {
+                $parentResourceMetadata = $this->decorated->create($resourceClass);
+            } catch (ResourceClassNotFoundException $resourceNotFoundException) {
+                // Ignore not found exception from decorated factories
+            }
+        }
+
+        if (!(class_exists($resourceClass) || interface_exists($resourceClass)) || !$resource = $this->extractor->getResources()[$resourceClass] ?? false) {
+            return $this->handleNotFound($parentResourceMetadata, $resourceClass);
+        }
+
+        $resource['description'] = $resource['description'] ?? $this->defaults['description'] ?? null;
+        $resource['iri'] = $resource['iri'] ?? $this->defaults['iri'] ?? null;
+        $resource['itemOperations'] = $resource['itemOperations'] ?? $this->defaults['item_operations'] ?? null;
+        $resource['collectionOperations'] = $resource['collectionOperations'] ?? $this->defaults['collection_operations'] ?? null;
+        $resource['graphql'] = $resource['graphql'] ?? $this->defaults['graphql'] ?? null;
```

Comment:
```suggestion
    foreach ($attributes as $attribute) {
        $resource[$attrbiute] = $resource[$attribiute] ?? $this->defaults[$attribute] ?? null
```
and array with all attributes, WDYT?

Reply:
its good idea, had to use some magic regex to convert values likse collectionOperations to collection_operations, seems like its working ;D